core/invoker/src/main/scala/whisk/core/invoker/ActionLogDriver.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.invoker
+
+import java.nio.charset.StandardCharsets
+
+import scala.Vector
+import scala.util.{ Failure, Success, Try }
+
+import spray.json._
+import spray.json.DefaultJsonProtocol
+import whisk.common.{ Logging, TransactionId }
+import whisk.core.entity._
+import whisk.core.entity.size.{ SizeInt, SizeString }
+import scala.collection.mutable.Buffer
+
+/**
+ * Represents a single log line as read from a docker log
+ */
+protected[invoker] case class LogLine(time: String, stream: String, log: String) {
+    def toFormattedString = f"$time%-30s $stream: ${log.trim}"
+    def dropRight(maxBytes: ByteSize) = {
+        val bytes = log.getBytes(StandardCharsets.UTF_8).dropRight(maxBytes.toBytes.toInt)
+        LogLine(time, stream, new String(bytes, StandardCharsets.UTF_8))
+    }
+}
+
+protected[invoker] object LogLine extends DefaultJsonProtocol {
+    implicit val serdes = jsonFormat3(LogLine.apply)
+}
+
+protected[invoker] trait ActionLogDriver {
+
+    // The action proxies inserts this line in the logs at the end of each activation for stdout/stderr
+    protected val LOG_ACTIVATION_SENTINEL = "XXX_THE_END_OF_A_WHISK_ACTIVATION_XXX"
+
+    /**
+     * Given the JSON driver's raw output of a docker container, convert it into our own
+     * JSON format. If asked, check for sentinel markers (which are not included in the output).
+     *
+     * Only parses and returns so much logs to fit into the LogLimit passed.
+     *
+     * @param logMsgs raw String read from a JSON log-driver written file
+     * @param requireSentinel determines if the processor should wait for a sentinel to appear
+     * @param limit the limit to apply to the log size
+     *
+     * @return Tuple containing (isComplete, isTruncated, logs)
+     */
+    protected def processJsonDriverLogContents(logMsgs: String, requireSentinel: Boolean, limit: ByteSize)(
```

Comment:
Not necessarily to be done in this PR: Should this get an `Iterator[String]` instead of all messages as a whole? That way we could read the file with `Source.fromFile(logFile).getLines` and thereby not load the whole file into memory?